Topic4Services/src/userServices/UserServiceImp.java

```diff
@@ -0,0 +1,44 @@
+package userServices;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class UserServiceImp implements UserService {
+
+	private Map<String, User> users;
+
+	public UserServiceImp() {
+		users = new HashMap<String, User>();
+	}
+
+	public void create(String id, String pass) {
+		User user = new User(id, pass);
+		users.put(id, user);

```

Comment:
You are not validating if an user exists prior to put it on the HashMap, so if a new user is added using the same id (what are you considering as the id?) the new one will overwrite the older